content/guides/spring-integration-lock/index.md

```diff
@@ -0,0 +1,366 @@
+     
+     ![img](images/sql_error.png)
+
+## Create a Lock Repository
+
+Once you import the necessary packages, you can start setting up your code. The first order of business is to create the lock repository beans that will be used to grab the locks later. 
+
+---
+
+### Redis
+
+With the Redis version, you need to create a `String` name to represent the `LockRegistry`. To learn more about how the `@Bean` provides access to an object [look here](https://docs.spring.io/spring-boot/docs/2.0.x/reference/html/using-boot-spring-beans-and-dependency-injection.html).
+
+```java
+private static final String LOCK_NAME = "lock";
+
+@Bean(destroyMethod = "destroy")
+public RedisLockRegistry redisLockRegistry(RedisConnectionFactory redisConnectionFactory) {
+    return new RedisLockRegistry(redisConnectionFactory, LOCK_NAME);
+}
+```
+
+---
+
+### JDBC
+
+Do the following:
+
+```java
+@Bean
+public DefaultLockRepository DefaultLockRepository(DataSource dataSource){
+    return new DefaultLockRepository(dataSource);
+}
+
+@Bean
+public JdbcLockRegistry jdbcLockRegistry(LockRepository lockRepository){
+    return new JdbcLockRegistry(lockRepository);
+}
+```
+
+## Setup A Controller
+
+In order to interact with our lock, you'll need to set up a way to hit your code from the outside world. The easiest way is to set up a `RestController` with a few endpoints for you to hit. 
+
+Note: To do this the proper Spring way, you are going to have to inject our service class. Instructions are in the next section, Setup the Service Class.
+
+```java
+@RestController
+@RequestMapping("/")
+public class MyController {
+    private final LockService lockService;
+
+    public MyController(LockService lockService) {
+        this.lockService = lockService;
+    }
+
+    @PutMapping("/lock")
+    public String lock(){
+        return lockService.lock();
+    }
+
+    @PutMapping("/properLock")
+    public String properLock(){
+        return lockService.properLock();
+    }
+
+    @PutMapping("/failLock")
+    public String failLock(){
+        lockService.failLock();
+        return "fail lock called, output in logs";
+    }
+}
+```
+
+It's OK if you don't understand everything that's happening here. The important part is to realize that the follow endpoints are set up: `/lock`, `/properLock`, and `/failLock` and that these endpoints will call functions in the service class that you will set up in the next section.
+
+## Setup the Service Class
+
+1. Create a simple interface to match the methods that we have in our controller, created in the above section.
+
+    ```java
+        public interface LockService {
+        String lock();
+        void failLock();
+        String properLock();
+    }
+    ```
+
+2. Set up the service class that will contain the logic that you want to lock. To do this, create a new class that implements `LockService`. Make sure it looks similar to the following:
+
+    ---
+
+    - Redis:
```

Comment:
The Redis and JDBC headers here are lead with a `-` but in the section above they are headers (ie ###) should they be the same formatting?

Reply:
Correct. The Redis and JDBC sections that start with a bullet (-) are subsections of a step in numbered list, whereas.  the two that you see as ### Redis and ### JDBC are Heading 3's that go with Heading 2, ## Create a Lock Repository.

Reply:
oh ok